packages/SwingSet/misc-tools/extract-xs-snapshot.js

```diff
@@ -53,6 +53,7 @@ if (!vatIDToExtract) {
   const { snapPos, hash } = info;
   const snapshot = snapStore.loadSnapshot(vatIDToExtract);
   const fn = `${vatIDToExtract}-${snapPos}-${hash}.xss`;
-  await fs.promises.writeFile(fn, snapshot);
-  console.log(`wrote snapshot to ${fn}`);
+  fs.promises
+    .writeFile(fn, snapshot)
+    .then(() => console.log(`wrote snapshot to ${fn}`));
```

Comment:
```suggestion
  E.when(
    fs.promises.writeFile(fn, snapshot),
    () => console.log(`wrote snapshot to ${fn}`));
```
Even when `then` is safe, we should still prefer `E.when` to `then` because of diagnostic info (track-turns deep stacks).

Reply:
This file is a one-off script that doesn't have any concurrent activity.  If you think it's still worth the effort, I'll make the change.

Reply:
Lots of code is written by imitation of other code. I do think this effort is worth it since this PR introduces the `.then` rather than just preserving one. Thanks.